### PR TITLE
Updated docker provider to latest version

### PR DESCRIPTION
**Pull request by pierskarsenbarg**
- Updated docker provider
- Added "classic" docker package (the last one of v3) to classic AWSx, so no real update is required there (unless you want one?)

The one thing I wasn't sure what to do about was [this value](https://github.com/pierskarsenbarg/pulumi-awsx/blob/1750308ef9e8666d299606fa300a1146f0dd7995/awsx/ecr/image.ts#L83) as there wasn't really a comparable option in the old version. Happy to take comments

Fixes: #1010

### CODE REVIEW

File: awsx/ecr/image.ts [hunk 1]
```diff
@@ -43,48 +43,49 @@ export function computeImageFromAsset(
...
+      args: dockerInputs.args,
+      cacheFrom: dockerInputs.cacheFrom
+        ? {
+            images: dockerInputs.cacheFrom,
+          }
+        : undefined,
+      platform: "linux/arm64",
+      target: dockerInputs.target,
+    },
+  };
```
Comment (danielrbradley) [inline awsx/ecr/image.ts:86]: This appears to be unused.

**Review by danielrbradley — COMMENTED** [review]
It looks like you're declaring `dockerImageArgs` but not using the value. 

On what to fill in the args with - I'd suggest we should just be transparently passing all arguments down to docker – we should change the definitions in the new awsx to reflect the new docker args exactly. This should be done programatically in the schema gen program.

This would cause the PR to technically be a breaking change, but I think this would be reasonable to ship in a minor version bump with changelog.

File: awsx/ecr/image.ts (hunk 1 above)
Comment (pierskarsenbarg) [reply awsx/ecr/image.ts]: Yeah that was me not being careful enough

File: sdk/nodejs/package.json [hunk 2]
```diff
@@ -17,9 +17,10 @@
     },
     "dependencies": {
         "@pulumi/aws": "^5.35.0",
-        "@pulumi/docker": "^3.6.1",
+        "@pulumi/docker": "^4.3.0",
         "@pulumi/pulumi": "^3.0.0",
         "@types/aws-lambda": "^8.10.23",
+        "docker-classic": "npm:@pulumi/docker@3.6.1",
```
[the remark this answers is not in the document]
Comment (pierskarsenbarg) [reply sdk/nodejs/package.json:23]: No nor me, but good to know

**Pull request closed without merging**